src/libs/actions/User.js

```diff
@@ -290,6 +259,69 @@ function clearContactMethodErrors(contactMethod, fieldName) {
     });
 }
 
+/**
+ * Adds a secondary login to a user's account
+ *
+ * @param {String} contactMethod
+ * @param {String} password
+ */
+function addNewContactMethodAndNavigate(contactMethod, password) {
+    Onyx.merge(ONYXKEYS.ACCOUNT, {...CONST.DEFAULT_ACCOUNT_DATA, isLoading: true});
+
+    const optimisticData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                partnerUserID: contactMethod,
+                validatedDate: '',
+                errorFields: {
+                    addedLogin: null,
+                },
+                pendingFields: {
+                    addedLogin: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+    }];
+    const successData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                partnerUserID: contactMethod,
+                validatedDate: '',
+                errorFields: {
+                    addedLogin: null,
+                },
+                pendingFields: {
+                    addedLogin: null,
+                },
+            },
+        },
+    }];
+    const failureData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                partnerUserID: contactMethod,
```

Comment:
Similarly, I think we can remove this line since it's already set in the optimisticData